database/rrddim.c

```diff
@@ -527,6 +527,11 @@ void rrddim_free_custom(RRDSET *st, RRDDIM *rd, int db_rotated)
             debug(D_RRD_CALLS, "Removing dimension '%s'.", rd->name);
             freez((void *)rd->id);
             freez(rd->cache_filename);
+#ifdef ENABLE_DBENGINE
+            if (rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+                freez(rd->state->metric_uuid);
```

Comment:
At this point we know the `uuid` and the object address. A call int othe dbengine could delete the entry from the map and free the memory - then we don't need to track the global list.

Reply:
(and the second call site obviously :) )